lib/cmsis_rtos_v1/cmsis_thread.c

```diff
@@ -80,7 +80,7 @@ osThreadId osThreadCreate(const osThreadDef_t *thread_def, void *arg)
 	atomic_dec((atomic_t *)&thread_def->instances);
 	stk_ptr = thread_def->stack_mem;
 	prio = cmsis_to_zephyr_priority(thread_def->tpriority);
-	k_thread_custom_data_set((void *)thread_def);
+	(&cm_thread[thread_def->instances])->custom_data = (void *)thread_def;
```

Comment:
Can't you just create the thread stopped, set the custom data with the existing API, and then start it?